clar2wasm/src/standard/standard.wat

```diff
@@ -2493,6 +2493,123 @@
         (local.get $len)
     )
 
+    (func $stdlib.uint-to-utf8 (param $lo i64) (param $hi i64) (result i32 i32)
+        (local $i i32) (local $j i32)
+        (local.set $j (local.tee $i (global.get $stack-pointer)))
+
+        ;; slow loop while $hi > 0
+        (if (i64.ne (local.get $hi) (i64.const 0))
+            (then
+                (loop $loop
+                    (call $stdlib.div-int128 (local.get $lo) (local.get $hi) (i64.const 10) (i64.const 0))
+                    ;; remainder on the stack
+                    drop ;; drop remainder_hi
+                    (local.set $lo (i64.add (i64.const 48))) ;; to ascii
+                    ;; to store a digit on 32 bits in big-endian, we write 4bytes of 0
+                    ;; and we replace the last one by the correct value
+                    (i32.store (local.get $i) (i32.const 0))
+                    (i64.store8 offset=3 (local.get $i) (local.get $lo))
```

Comment:
Can we just shift and store here instead of doing two stores?

Reply:
yup, done